packages/gestalt/src/AvatarGroup.js

```diff
@@ -0,0 +1,186 @@
+// @flow strict
+import { forwardRef, useState, type Node } from 'react';
+import PropTypes from 'prop-types';
+import Box from './Box.js';
+import TapArea, { type OnTapType } from './TapArea.js';
+
+import {
+  AddCollaboratorsButton,
+  CollaboratorAvatar,
+  CollaboratorsCount,
+  SizeProptype,
+  type Size,
+} from './AvatarGroupSubcomponents.js';
+import Flex from './Flex.js';
+import { type CollaboratorDataType } from './Avatar.js';
+
+type Role = 'link' | 'button';
+
+type Props = {|
+  accessibilityLabel: string,
+  accessibilityControls?: string,
+  accessibilityExpanded?: boolean,
+  accessibilityHaspopup?: boolean,
+  addCollaborators?: boolean,
+  collaborators: $ReadOnlyArray<CollaboratorDataType>,
+  href?: string,
+  onClick?: OnTapType,
+  role?: Role,
+  size?: Size,
+|};
+
+type UnionRefs = HTMLDivElement | HTMLAnchorElement;
+
+const AvatarGroupWithForwardRef: React$AbstractComponent<Props, UnionRefs> = forwardRef<
+  Props,
+  UnionRefs,
+>(function AvatarGroup(
+  {
+    accessibilityLabel,
+    accessibilityControls,
+    accessibilityExpanded,
+    accessibilityHaspopup,
+    addCollaborators,
+    collaborators,
+    href,
+    onClick,
+    role,
+    size = 'fit',
+  },
+  ref,
+): Node {
+  const [hovered, setHovered] = useState(false);
+
+  const MAX_COLLABORATOR_AVATARS = 3;
+
+  const isMdSize = size === 'md';
+
+  const isFitSize = size === 'fit';
+
+  const isMdOrFitSize = isMdSize || isFitSize;
+
+  const isDisplayOnly = !role;
+
+  const isAboveMaxCollaborators = collaborators.length > MAX_COLLABORATOR_AVATARS;
+
+  const addCollaboratorsCount = isMdOrFitSize && isAboveMaxCollaborators;
+
+  const addCollaboratorsButton = (isMdOrFitSize && !isDisplayOnly && addCollaborators) ?? false;
+
+  const slicedCollaborators = collaborators.slice(
```

Comment:
This variable name doesn't tell us anything we can't see by the `slice` call. Perhaps something like `collaboratorsToDisplay`, `displayedCollaborators`, something like that?

Reply:
DONE